sympy/core/compatibility.py

```diff
@@ -676,6 +677,46 @@ def ordered(seq, keys=None, default=True, warn=False):
         yield from d[k]
         d.pop(k)
 
+class OpDispatcher:
+    """
+    Dispatcher for operations, e.g. add, mul, pow, ...
+
+    .. notes::
+       This approach is experimental, and can be replaced or deleted in the future.
+       See https://github.com/sympy/sympy/pull/19463.
+
+    """
+    def __init__(self, name, fallback, doc=None):
+        self.name = name
+        self.fallback = fallback
+        self.doc = doc
+        self.handlermethod = "_%s_handler" % name
+        self.dispatcher = Dispatcher(name)
```

Comment:
Ok, I think I understand my confusion here. `OpDispatcher` _is_ a `Dispatcher`, it is used as
```python
    @power.register(NewBase, NewBase)
    def _(a, b):
        return NewPow(a, b)
```

But `AssocOpDispatcher` is _not_ a `Dispatcher`, because it is used as
```python
    @add.register_priority(NewBase, NewBase)
    def _(a, b):
        return NewBase

    
```

So I think you should put `OpDispatcher` back the way it was, and make the changes only to `AssocOpDispatcher`

Reply:
Thanks! Seems that's the best way.